### PR TITLE
Common Readme Names

### PR DESCRIPTION
**Pull request by scbedd**
@anmeng10101 @sima-zhu This is a refactor of how we store documents under `docs-ref-services/`

The problem was that on a given moniker, an overview page will display the readme for the package right? However, if you click the dropdown and change from preview->latest (or vice-versa) you are IMMEDIATELY hit with a 404 and redirect back to the previously selected moniker.

This is due to the fact that the readmes themselves are named differently! `latest` maps to `blob-readme.md` while `preview` would map to `blob-readme-pre.md`.

By placing them into different folders, we can name them the SAME and allow enable moniker swaps!

### CODE REVIEW

File: .openpublishing.redirection.json [hunk 1]
```diff
@@ -50475,5 +50475,455 @@
       "redirect_url": "/java/api/tests.integration.com.microsoft.azure.sdk.iot.transport._https_batch_message_i_t.UTF8().raw.page.js",
       "redirect_document_id": false
     },
+    {
+      "source_path": "docs-ref-services/ai-anomalydetector-readme-pre.md",
```
Comment (sima-zhu) [inline .openpublishing.redirection.json:50479]: Is this another manual update place for newly onboard services?
What's the purpose of having this?

Comment (scbedd) [reply .openpublishing.redirection.json:50479]: Let's have a quick call!

**Comment by anmeng10101** [comment]
@scbedd I am not sure if we still need to build the conceptual articles in this repo. From the pr to remove the ConceptualToc configuration, feels like it is targeting to managed all SDK related conceptual articles in https://github.com/MicrosoftDocs/azure-dev-docs-pr. 

I will recommend to check with @rloutlaw first.

**Comment by anmeng10101** [comment]
@scbedd  This brilliant idea!! 

2 questions:

1. In SDK release pipeline, the README.md file will be copy to all three folders?
2. [update_toc.py](https://github.com/Azure/azure-docs-sdk-java/blob/feature/similar-named-documents/update_toc.py) also need update before we could run the CI?

**Review by sima-zhu — APPROVED** [review]

**Pull request merged**